plugins/experimental/remap_stats/remap_stats.cc

```diff
@@ -160,81 +140,80 @@ handle_post_remap(TSCont cont, TSEvent event ATS_UNUSED, void *edata)
   return 0;
 }
 
-#define CREATE_STAT_NAME(s, h, b) snprintf(s, MAX_STAT_LENGTH, "plugin.%s.%s.%s", PLUGIN_NAME, h, b)
+static std::string
+create_stat_name(const std::string &h, const std::string &b)
+{
+  std::string res(std::string("plugin.") + PLUGIN_NAME + '.' + h + '.' + b);
```

Comment:
This is more efficiently done with
```
std::string_view PREFIX{"plugin."};
static const size_t pn_len = strlen(PLUGIN_NAME);
std::string res;
res.reserve(PREFIX.size() + pn_len + 1+ h.size() + 1 + b.size() + 1);
res = PREFIX;
res += PLUGIN_NAME;
res += '.';
res += h;
res += '.';
res += b;
// Or this for the daring - 
// snprintf(res.data(), res.capacity(), "%s.%s.%s.%s", PREFIX.data(), PLUGIN_NAME, h.c_str(), b.c_str());
return std::move(res);
}
```
Yes, it's a lot more code, but it does at most one allocation. It's shorter if you abuse `snprintf`.

Reply:
Ok, I'll go with the longer version. :+1:

Reply:
Note that in C++17, `std::string::data()` was changed to return `char*` instead of `const char*` precisely to allow this kind of use of `snprintf` and equivalents.

Reply:
If you prefer the version with `snprintf` I can go with that.

Reply:
gcc9 may complain about the unnecessary call to std::move in the return as res is already local scope.  Depends on the compiler debug levels I'm guessing.

Reply:
@traeak I've been doing that quite a bit lately, using gcc 9, and haven't had it complain. I think it's not until C++20 that the compiler is allowed to move returned local variables.